internal/services/autopi/hardware_template_service.go

```diff
@@ -151,6 +152,24 @@ func (a *hardwareTemplateService) ApplyHardwareTemplate(ctx context.Context, req
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to unassociate template")
 		}
+
+		vehicleLoggers := new([]services.AutoPiVehicleLogger)
+		err = a.ap.GetVehicleLoggers(autoPi.Vehicle, vehicleLoggers)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("failed to get vehicle loggers for vehicleID %d", autoPi.Vehicle.ID))
+		} else {
+			for _, logger := range *vehicleLoggers {
+				if logger.Parent == 0 {
+					loggerType := logger.LoggerType
+					loggerID := logger.ID
+					err := a.ap.DeleteVehicleLogger(loggerType, loggerID)
```

Comment:
just inline loggerType and loggerID from above two lines, eg. `(logger.LoggerType, logger.ID)`